haskell/private/haskell_impl.bzl

```diff
@@ -179,14 +217,41 @@ def _haskell_binary_common_impl(ctx, is_test):
         bin_info = bin_info,
     )
 
+    executable = binary
+
+    if inspect_coverage:
+        binary_path = ctx.workspace_name + "/" + binary.short_path
```

Comment:
Use path building tools, such as `paths.join`. I don't know how this affect portability, but that's the way to do it.